src/compiler/program.ts

```diff
@@ -65,7 +65,14 @@ namespace ts {
                 : { resolvedModule: undefined, failedLookupLocations };
         }
         else {
-            return loadModuleFromNodeModules(moduleName, containingDirectory, host);
+            let result = loadModuleFromNodeModules(moduleName, containingDirectory, host, compilerOptions);
+            if(result && result.resolvedModule && result.resolvedModule.resolvedFileName){
+                if(!fileExtensionIs(result.resolvedModule.resolvedFileName, ".ts")){
+                    // Loaded a non-TypeScript file from a node modules search
+                    result.resolvedModule.isJsFileFromNodeSearch = true;

```

Comment:
Agreed. And after talking to Vlad and reviewing the code, I can likely also just use the flag he already has (`isExternalLibrary`).